partiql-lang/src/test/kotlin/org/partiql/lang/eval/visitors/SubqueryCoercionVisitorTransformTests.kt

```diff
@@ -0,0 +1,113 @@
+                original = """ (1, 2, 3) < (SELECT n FROM [1,2,3] AS n) """,
+                expected = """ (1, 2, 3) < (SELECT n FROM [1,2,3] AS n)"""
+            ),
+            TransformTestCase(
+                name = "Coerce lhs of IN, but not rhs",
+                original = """ (SELECT max(n) FROM <<1,2,3>> AS n) IN (SELECT n FROM [1,2,3] AS n) """,
+                expected = """ coll_to_scalar(SELECT max(n) FROM <<1,2,3>> AS n) IN (SELECT n FROM [1,2,3] AS n) """
+            ),
+            TransformTestCase(
+                name = "Coerce subquery that is the WHERE expression",
+                original = """
+                    SELECT x.b
+                    FROM << {'a': 1, 'b': 10}, {'a': 3, 'b': 30} >>  AS x
+                    WHERE (SELECT max(n) = x.a FROM [1,2,3] AS n) 
+                """.trimIndent(),
+                expected = """
+                    SELECT x.b
+                    FROM << {'a': 1, 'b': 10}, {'a': 3, 'b': 30} >>  AS x
+                    WHERE coll_to_scalar(SELECT max(n) = x.a FROM [1,2,3] AS n) 
+                """.trimIndent()
+            ),
+            TransformTestCase(
+                name = "Coerce SELECT when the only item in another SELECT",
+                original = """ SELECT (SELECT max(n) FROM [1,2,3] AS n) AS m FROM 0 """,
+                expected = """ SELECT coll_to_scalar(SELECT max(n) FROM [1,2,3] AS n) AS m FROM 0 """
+            ),
+            TransformTestCase(
+                name = "Coerce SELECT when among items in another SELECT",
+                original = """
+                     SELECT x.b, 
+                            (SELECT y.c
+                             FROM << {'a': 1, 'c': 100}, {'a': 3, 'c': 300} >> AS y
+                             WHERE y.a = x.a ) AS c2
+                     FROM << {'a': 1, 'b': 10}, {'a': 3, 'b': 30} >>  AS x
+                """.trimIndent(),
+                expected = """
+                    SELECT x.b, 
+                           coll_to_scalar(
+                            SELECT y.c
+                            FROM << {'a': 1, 'c': 100}, {'a': 3, 'c': 300} >> AS y
+                            WHERE y.a = x.a ) AS c2
+                    FROM << {'a': 1, 'b': 10}, {'a': 3, 'b': 30} >>  AS x
+                """.trimIndent()
+            ),
+            TransformTestCase(
+                name = "Do not coerce data sources in FROM",
+                original = """ 
+                    SELECT n2.n1, m2.m
+                    FROM (SELECT n1 FROM <<1,2,3>> AS n1) AS n2,
+                         (SELECT min(m1) AS m FROM [4,5] AS m1) AS m2
+                    WHERE 2 * n2.n1 = m2.m
+                """.trimIndent(),
+                expected = """
+                     SELECT n2.n1, m2.m
+                     FROM (SELECT n1 FROM <<1,2,3>> AS n1) AS n2,
+                          (SELECT min(m1) AS m FROM [4,5] AS m1) AS m2
+                     WHERE 2 * n2.n1 = m2.m
+                """.trimIndent()
```

Comment:
Not a blocker for this PR: would be nice if we can create an issue in [`partiql-tests`](https://github.com/partiql/partiql-tests) for adding these tests to `partiql-tests`:
- https://github.com/partiql/partiql-tests/blob/main/partiql-tests-data/eval/spec-tests.ion#L612-L636
- https://github.com/partiql/partiql-tests/blob/main/partiql-tests-data/eval-equiv/spec-tests.ion#L366-L453

Thanks to @alancai98 for the pointers.

Reply:
See https://github.com/partiql/partiql-tests/issues/70